mrp_production_estimated_cost/models/mrp_production.py

```diff
@@ -0,0 +1,271 @@
+        if not production.analytic_account_id:
+            raise exceptions.Warning(
+                _('You must define one Analytic Account for this MO: %s') %
+                (production.name))
+        return {
+            'name': name,
+            'mrp_production_id': production.id,
+            'workorder': workorder and workorder.id or False,
+            'account_id': self.analytic_account_id.id,
+            'journal_id': journal.id,
+            'user_id': self.env.uid,
+            'date': analytic_line_obj._get_default_date(),
+            'product_id': product and product.id or False,
+            'unit_amount': qty,
+            'amount': 0,
+            'product_uom_id': product.uom_id.id,
+            'general_account_id': general_account.id,
+            'estim_std_cost': -qty * (std_cost or
+                                      product.manual_standard_cost),
+            'estim_avg_cost': -qty * (avg_cost or product.standard_price),
+        }
+
+    @api.model
+    def _create_material_estimated_cost(self, prod, product_line):
+        if not product_line.product_id:
+            raise exceptions.Warning(
+                _("One consume line has no product assigned."))
+        journal = self.env.ref('mrp.analytic_journal_materials', False)
+        name = '%s-%s' % (prod.name, product_line.work_order.name or '')
+        product = product_line.product_id
+        qty = product_line.product_qty
+        vals = self._prepare_estimated_cost_analytic_line(
+            journal, name, prod, product, workorder=product_line.work_order,
+            qty=qty)
+        return self.env['account.analytic.line'].create(vals)
+
+    @api.model
+    def _create_pre_operation_estimated_cost(self, prod, wc, workorder):
+        if workorder.time_start:
+            product = workorder.workcenter_id.pre_op_product
+            journal = self.env.ref('mrp.analytic_journal_machines', False)
+            name = (_('%s-%s Pre-operation') %
+                    (prod.name, workorder.workcenter_id.name))
+            vals = self._prepare_estimated_cost_analytic_line(
+                journal, name, prod, product, workorder=workorder,
+                qty=workorder.time_start)
+            return self.env['account.analytic.line'].create(vals)
+
+    @api.model
+    def _create_post_operation_estimated_cost(self, prod, wc, workorder):
+        if workorder.time_stop:
+            product = workorder.workcenter_id.post_op_product
+            journal = self.env.ref('mrp.analytic_journal_machines', False)
+            name = (_('%s-%s Post-operation') %
+                    (prod.name, workorder.workcenter_id.name))
+            vals = self._prepare_estimated_cost_analytic_line(
+                journal, name, prod, product, workorder=workorder,
+                qty=workorder.time_stop)
+            return self.env['account.analytic.line'].create(vals)
+
+    @api.model
+    def _create_worcenter_cycles_estimated_cost(self, prod, wc, workorder):
```

Comment:
It is really a misstyping do not know if it has to be really fixed but just in case there shows `worcenter` instead of `workcenter`